src/g_game.c

```diff
@@ -295,7 +295,7 @@ static int G_NextWeapon(int direction)
 
     if (i == arrlen(weapon_order_table))
     {
-        return wp_nochange;
+        I_Error("G_NextWeapon: Invalid weapon type %d", (int)weapon);
```

Comment:
Why do we need `I_Error` here? There are known demos with this error, which are played to the end on the current version.

Reply:
You mean these demos here https://github.com/fabiangreffrath/woof/issues/1245 ? I don't think this is the same case. They trigger bugs caused by the Boom auto-switch feature. This is about manually selecting prev/next weapon, which should never trigger this bug.